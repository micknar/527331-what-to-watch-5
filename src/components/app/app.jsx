import React from "react";
import {BrowserRouter, Switch, Route} from "react-router-dom";
import PropTypes from "prop-types";
import Main from "../main/main";
import SignIn from "../sign-in/sign-in";
import MyList from "../my-list/my-list";
import FilmPage from "../film-page/film-page";
import AddReview from "../add-review/add-review";
import Player from "../player/player";

const App = (props) => {
  const {films, promoFilm} = props;

  return (
    <BrowserRouter>
      <Switch>
        <Route
          exact
          path="/"
          render={({history}) => (
            <Main
              promoFilm={promoFilm}
              films={films}
              onFilmCardClick={(id) => history.push(`/films/${id}`)}
            />
          )}
        />

        <Route exact path="/login">
          <SignIn />
        </Route>

        <Route
          exact
          path="/mylist"
          render={({history}) => (
            <MyList
              films={films}
              onFilmCardClick={(id) => history.push(`/films/${id}`)}
            />
          )}
        />

        <Route
          exact
          path="/films/:id"
          render={({history}) => (
            <FilmPage
              films={films}
              onFilmCardClick={(id) => history.push(`/films/${id}`)}
            />
          )}
        />

        <Route exact path="/films/:id/review">
          <AddReview films={films} />
        </Route>

        <Route exact path="/player/:id">
          <Player />
        </Route>
      </Switch>
    </BrowserRouter>
  );
};

App.propTypes = {
  films: PropTypes.array.isRequired,
  promoFilm: PropTypes.object.isRequired,
};

export default App;
