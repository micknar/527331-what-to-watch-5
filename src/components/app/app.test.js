import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app";

jest.mock(`../main/main`, () => `Main`);
jest.mock(`../sign-in/sign-in`, () => `SignIn`);
jest.mock(`../my-list/my-list`, () => `MyList`);
jest.mock(`../film-page/film-page`, () => `FilmPage`);
jest.mock(`../add-review/add-review`, () => `AddReview`);
jest.mock(`../fullscreen-player/fullscreen-player`, () => `FullscreenPlayer`);
jest.mock(`../private-route/private-route`, () => `PrivateRoute`);
jest.mock(`../loading-page/loading-page`, () => `LoadingPage`);
jest.mock(`../error-page/error-page`, () => `ErrorPage`);
jest.mock(`../not-found-page/not-found-page`, () => `NotFoundPage`);

describe(`Should App render correctly`, () => {
  it(`With loading success`, () => {
    const tree = renderer
      .create(
          <App
            isFilmsLoading={false}
            isPromoLoading={false}
            isLoadingError={false}
            isCheckingAuth={false}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With checking authorization`, () => {
    const tree = renderer
      .create(
          <App
            isFilmsLoading={false}
            isPromoLoading={false}
            isLoadingError={false}
            isCheckingAuth={true}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When data loading`, () => {
    const tree = renderer
      .create(
          <App
            isFilmsLoading={true}
            isPromoLoading={true}
            isLoadingError={false}
            isCheckingAuth={false}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With loading error`, () => {
    const tree = renderer
      .create(
          <App
            isFilmsLoading={false}
            isPromoLoading={false}
            isLoadingError={true}
            isCheckingAuth={false}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
