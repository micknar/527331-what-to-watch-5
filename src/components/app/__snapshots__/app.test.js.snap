// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should App render correctly When data loading 1`] = `<LoadingPage />`;

exports[`Should App render correctly With checking authorization 1`] = `<LoadingPage />`;

exports[`Should App render correctly With loading error 1`] = `<ErrorPage />`;

exports[`Should App render correctly With loading success 1`] = `
<Main
  onCardClick={[Function]}
/>
`;
